File: WEB-INF/classes/Quiz.java
import java.nio.charset.StandardCharsets;
import org.json.JSONObject;

public class Quiz extends AClass {
    private String name;
    private String category_name = "";
    private String description = "";
    private byte[] media_id = null;

    public Quiz() {
        super("quiz");
    }

    public Quiz(String constructorParams) {
        super("quiz");
        String[] keyvaluePairs = constructorParams.split(",,,");
        for (String pair : keyvaluePairs) {
            String[] keyvaluePair = pair.split(":==");
            if(keyvaluePair.length > 1){
                switch (keyvaluePair[0]) {
                    case "name":
                        this.name = keyvaluePair[1];
                        break;
                    case "category_name":
                        this.category_name = keyvaluePair[1];
                        break;
                    case "description":
                        this.description = keyvaluePair[1];
                        break;
                    case "media_id":
                        this.media_id = keyvaluePair[1].getBytes();
                        break;
                }
            }
            
        }
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCategoryName() {
        return this.category_name;
    }

    public void setCategoryName(String category_name) {
        this.category_name = category_name;
    }

    public String getDescription() {
        return this.description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public byte[] getMediaId() {
        return this.media_id;
    }

    public void setMediaId(byte[] media_id) {
        this.media_id = media_id;
    }

    @Override
    public JSONObject serialize() {
        JSONObject jo = null;

        try {
            jo = new JSONObject();
            jo.put("name", this.name);
            jo.put("category_name", this.category_name);
            jo.put("description", this.description);
            jo.put("media_id",
                    (this.media_id != null ? new String(this.media_id, StandardCharsets.UTF_8) : this.media_id));
        } catch (Exception e) {
            e.printStackTrace();
        }

        return jo;
    }

}

File: QuizAppTest/Utilities/Quiz.cs
using System.Text.Json.Nodes;

namespace QuizApp.Utilities
{
    public class Quiz : AClass
    {
        private string name = "";
        private string category_name = "";
        private string description = "";
        private byte[]? media_id;

        public Quiz() : base("quiz")
        {
            
        }

        public Quiz(string constructorParams) : base("quiz")
        {
            string[] keyvaluePairs = constructorParams.Split(",,,");
            foreach (string pair in keyvaluePairs)
            {
                string[] keyvaluePair = pair.Split(":==");
                if (keyvaluePair.Length > 1)
                {
                    switch (keyvaluePair[0])
                    {
                        case "name":
                            this.name = keyvaluePair[1];
                            break;
                        case "category_name":
                            this.category_name = keyvaluePair[1];
                            break;
                        case "description":
                            this.description = keyvaluePair[1];
                            break;
                        case "media_id":
                            this.media_id = Convert.FromHexString(keyvaluePair[1]);
                            break;
                    }
                }

            }
        }

        public string getName()
        {
            return this.name;
        }

        public void setName(string name)
        {
            this.name = name;
        }

        public string getCategoryName()
        {
            return this.category_name;
        }

        public void setCategoryName(string category_name)
        {
            this.category_name = category_name;
        }

        public string getDescription()
        {
            return this.description;
        }

        public void setDescription(string description)
        {
            this.description = description;
        }

        public byte[]? getMediaId()
        {
            return this.media_id;
        }

        public void setMediaId(byte[] media_id)
        {
            this.media_id = media_id;
        }
    public override JsonObject serialize()
        {
            JsonObject jo = new JsonObject
            {
                ["name"] = this.name,
                ["category_name"] = this.category_name,
                ["description"] = this.description,
                ["media_id"] = (this.media_id != null ? BitConverter.ToString(this.media_id).Replace("-", "") : "")
            };

            return jo;
        }
    }
}
